qmetaobject/src/qmetatype.rs

```diff
@@ -71,12 +72,14 @@ fn register_metatype_common<T: QMetaType>(
         let name = name.as_ptr();
         let type_id = cpp!(unsafe [name as "const char*", size as "int", deleter_fn as "QMetaType::Deleter",
                 creator_fn as "QMetaType::Creator", destructor_fn as "QMetaType::Destructor",
-                constructor_fn as "QMetaType::Constructor", gadget_metaobject as "const QMetaObject*"] -> i32 as "int" {
-            QMetaType::TypeFlags extraFlag(gadget_metaobject ? QMetaType::IsGadget : 0);
-            return QMetaType::registerType(gadget_metaobject ? gadget_metaobject->className() : name, deleter_fn, creator_fn, destructor_fn,
+                constructor_fn as "QMetaType::Constructor", metaobject as "const QMetaObject*",
+                is_enum as "bool"] -> i32 as "int" {
+            auto type_flag = is_enum ? QMetaType::IsEnumeration : QMetaType::IsGadget;
+            QMetaType::TypeFlags extraFlag(metaobject ? type_flag : 0);
+            return QMetaType::registerType(metaobject ? metaobject->className() : name, deleter_fn, creator_fn, destructor_fn,
                 constructor_fn, size,
```

Comment:
here, `size` should match the  the actual size behind the pointer which is passed to the QVarient.
So if you're gonna always use u32,  it should always be sizeof(u32)  for enums

Reply:
Seems legit.
So either I stick to `u32` and fix it here, or I add the representation to the `QEnum`.
Got it, make sense.